fbw-a32nx/src/systems/fmgc/src/guidance/vnav/descent/TodGuidance.ts

```diff
@@ -44,61 +47,68 @@ export class TodGuidance {
 
   update(deltaTime: number) {
     this.updateTdReached(deltaTime);
-    this.updateTdPause(deltaTime);
+    if (NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
+      this.updateTdPause(deltaTime);
+    }
   }
 
   updateTdPause(deltaTime: number) {
-    if (this.cooldown <= 0 && NXDataStore.get('PAUSE_AT_TOD', 'DISABLED') === 'ENABLED') {
-      // Only watching if T/D pause untriggered + between flight phase CLB and CRZ
+    // Only armed if all conditions met
+    this.tdArmed =
+      this.cooldown <= 0 &&
+      !this.tdPaused &&
+      this.observer.get().flightPhase >= FmgcFlightPhase.Climb &&
+      this.observer.get().flightPhase <= FmgcFlightPhase.Cruise &&
+      Simplane.getAutoPilotAirspeedManaged();
+
+    if (SimVar.GetSimVarValue('L:A32NX_PAUSE_AT_TOD_ARMED', 'bool') !== this.tdArmed)
+      SimVar.SetSimVarValue('L:A32NX_PAUSE_AT_TOD_ARMED', 'bool', this.tdArmed);
+
```

Comment:
You could do this without reading the simvar, save the previous state of this.tdArmed before updating it and only set the simvar if it has changed. Minor optimization but should also be quick to do.

Reply:
Oh I must have forgotten to upstream the commit that changed this. I'm using LocalSimVar that essentially does this (in an abstracted function).